components/builder-api/src/server/resources/profile.rs

```diff
@@ -116,7 +122,20 @@ fn generate_access_token(req: HttpRequest<AppState>) -> HttpResponse {
     request.set_token(token);
 
     match route_message::<AccountTokenCreate, AccountToken>(&req, &request) {
-        Ok(account_token) => HttpResponse::Ok().json(account_token),
+        Ok(account_token) => {
+            // Memcache supports multiple tokens but to preserve legacy behavior
+            // we must purge any existing tokens AFTER generating new ones
+            match do_get_access_tokens(&req, account_id) {
+                Ok(access_tokens) => {
+                    let mut memcache = req.state().memcache.borrow_mut();
+                    for token in access_tokens.get_tokens() {
+                        memcache.delete_key(token.get_token())
```

Comment:
still not correct.. we need to delete the old token, not the new one... so we have to get the access token prior to the DB call... then delete the old token from cache after the DB call.

Reply:
oh I see now - LOL that last change would have been a fun failure.